dot/core/service.go

```diff
@@ -496,6 +497,31 @@ func (s *Service) InsertKey(kp crypto.Keypair) {
 	s.keys.Insert(kp)
 }
 
+// HasKey returns true if given hex encoded public key string is found in keystore, false otherwise, error if there
+//  are issues decoding string
+func (s *Service) HasKey(pubKeyStr string, keyType string) (bool, error) {
+	keyBytes, err := common.HexToBytes(pubKeyStr)
+	if err != nil {
+		return false, err
+	}
+	cKeyType := keystore.DetermineKeyType(keyType)
+	var err2 error
```

Comment:
why do you need `err2`?

Reply:
I thought I had an issue with shadowing, but I see now that I didn't...  Updated.